packages/protocol/scripts/foundry/create_and_migrate_anvil_l2_devchain.sh

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# Read environment variables and constants
+source $PWD/scripts/foundry/constants.sh
+
+# Generate and run L1 devchain
+echo "Generating and running L1 devchain before activating L2..."
+source $PWD/scripts/foundry/create_and_migrate_anvil_devchain.sh
+
+# Stop L1 devchain
+source $PWD/scripts/foundry/stop_anvil.sh
+
+# Duplicate L1 state
+cp $TMP_FOLDER/$L1_DEVCHAIN_FILE_NAME $TMP_FOLDER/$L2_DEVCHAIN_FILE_NAME
+
+# Start L2 devchain from L1 state and apply L2 migrations
+echo "Anvil L2 state will be saved to $TMP_FOLDER/$L2_DEVCHAIN_FILE_NAME"
```

Comment:
CI has a bug here:

```
Killed Anvil
Anvil L2 state will be saved to /runner/_work/celo-monorepo/celo-monorepo/packages/protocol/.tmp/l2-devchain.json
Activating L2 on the devchain...
Error: 
error sending request for url (http://127.0.0.1:8546/): error trying to connect: tcp connect error: Connection refused (os error 111)

Context:
- Error #0: error trying to connect: tcp connect error: Connection refused (os error 111)
- Error #1: tcp connect error: Connection refused (os error 111)
- Error #2: Connection refused (os error 111)
```

Source: [CI](https://github.com/celo-org/celo-monorepo/actions/runs/9837180487/job/27154422476?pr=11107#step:19:7130)

Reply:
The bug seems to be resolved as of this [CI run](https://github.com/celo-org/celo-monorepo/actions/runs/9838934169/job/27159898549?pr=11107) in this commit cd6ca0544b8d70a4c18f7393d3e3965829485260.